src/modules/projects/components/ProjectOutgoingReferralDetailsSubForm.tsx

```diff
@@ -0,0 +1,90 @@
+import { Box } from '@mui/system';
+import { useMemo } from 'react';
+
+import LoadingSkeleton from '@/components/elements/LoadingSkeleton';
+import DynamicForm from '@/modules/form/components/DynamicForm';
+import { useDynamicFormHandlersForRecord } from '@/modules/form/hooks/useDynamicFormHandlersForRecord';
+import useFormDefinition from '@/modules/form/hooks/useFormDefinition';
+import { SubmitFormAllowedTypes } from '@/modules/form/types';
+import { AlwaysPresentLocalConstants } from '@/modules/form/util/formUtil';
+import { RecordFormRole } from '@/types/gqlTypes';
+
+interface Props {
+  enrollmentId: string; // HoH enrollment that is being referred
+  destinationProjectId: string; // project being referred to
+  onSuccess: VoidFunction;
+}
+
+/**
+ * Renders a dynamic form for collecting additional referral details
+ * based on the receiving project
+ */
+const ProjectOutgoingReferralDetailsSubForm: React.FC<Props> = ({
+  destinationProjectId,
+  enrollmentId,
+  onSuccess,
+}) => {
+  const { formDefinition, loading: formDefinitionLoading } = useFormDefinition({
+    role: RecordFormRole.Referral,
+    // form selection is based on the project that is being referred to
+    projectId: destinationProjectId,
+  });
+  const localConstants = useMemo(
+    () => ({
+      projectId: destinationProjectId,
+      ...AlwaysPresentLocalConstants,
+    }),
+    [destinationProjectId]
+  );
+  const hookArgs = useMemo(() => {
+    return {
+      formDefinition,
+      record: {} as SubmitFormAllowedTypes,
+      inputVariables: {
+        projectId: destinationProjectId,
+        enrollmentId,
+      },
+      onCompleted: () => onSuccess(),
+    };
+  }, [formDefinition, destinationProjectId, enrollmentId, onSuccess]);
+
+  const { initialValues, errors, onSubmit, submitLoading } =
+    useDynamicFormHandlersForRecord(hookArgs);
+
+  if (!formDefinitionLoading && !formDefinition) {
+    throw new Error(
+      `Failed to find referral form definition for project ${destinationProjectId}`
+    );
+  }
+
+  return (
+    <Box>
+      {formDefinitionLoading && (
+        <LoadingSkeleton width={'100%'} count={1} sx={{ my: 2 }} />
+      )}
+      {formDefinition && (
+        <Box
+          // unset top-level card styling for form sections
+          sx={{ '.HmisForm-card': { px: 0, pt: 1, pb: 0, border: 'unset' } }}
```

Comment:
This is overriding styles within DynamicForm, right? Do you think it would be worth it to pull this into the component and handle it with a variant prop? \<DynamicForm variant='subform'/\>

Reply:
Good point, there were actually a few places that we did this. The others are modals. I made the variant `without_top_level_cards` so it's more descriptive, since it applies to some modals and sub-forms.